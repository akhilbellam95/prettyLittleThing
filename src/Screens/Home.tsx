import { Pressable, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import Products from '../Components/Products';
import { ParamListBase, useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

const Home = (): JSX.Element => {
  const navigation = useNavigation<NativeStackNavigationProp<ParamListBase>>();

  return (
    <View style={styles.container}>
      <View style={{ flex: 1 }}>
        <Products />
      </View>
      <View style={styles.cartButtonBox}>
        <Pressable
          testID='go-to-cart'
          style={styles.cartButton}
          onPress={() => navigation.navigate('Cart')}
        >
          <Text style={styles.goToCart}>Go to Cart</Text>
        </Pressable>
      </View>
    </View>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'beige',
    alignItems: 'center',
  },
  cartButton: {
    backgroundColor: '#FDBEDB',
    borderColor: 'black',
    borderWidth: 1,
    height: 50,
    justifyContent: 'center',
    paddingHorizontal: 30,
    marginVertical: 10,
    borderRadius: 10,
  },
  cartButtonBox: {
    backgroundColor: 'white',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  goToCart: {
    textAlign: 'center',
    fontWeight: 'bold',
  },
});
