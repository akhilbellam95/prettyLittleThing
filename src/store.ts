// eslint-disable-next-line prettier/prettier
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './Features/Cart/cartSlice';
import productReducer from './Features/Product/productSlice';

export const store = configureStore({
  reducer: {
    cart: cartReducer,
    product: productReducer,
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;

export type RootState = ReturnType<typeof store.getState>;
