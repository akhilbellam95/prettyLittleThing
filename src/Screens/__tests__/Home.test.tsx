import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import Home from '../Home';
import { Provider } from 'react-redux';
import { store } from '../../store';
import { act } from 'react-test-renderer';

const mockedNavigation = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({ navigate: mockedNavigation }),
}));

const navigation = { navigate: jest.fn() };

describe('Home component testing', () => {
  const tree = (
    <Provider store={store}>
      <Home />
    </Provider>
  );

  it('renders correctly', () => {
    const snap: unknown = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });

  test('When user clisk on Go to cart', async () => {
    const { getByTestId } = render(tree);

    await act(() => {
      fireEvent.press(getByTestId('go-to-cart'));

      expect(navigation.navigate).toBeTruthy();
    });
  });
});
