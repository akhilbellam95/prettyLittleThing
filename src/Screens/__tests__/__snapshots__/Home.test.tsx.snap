// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component testing renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "beige",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <RCTScrollView
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FDBEDB",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "paddingHorizontal": 30,
        }
      }
      testID="go-to-cart"
    >
      <Text
        style={
          {
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Go to Cart
      </Text>
    </View>
  </View>
</View>
`;
